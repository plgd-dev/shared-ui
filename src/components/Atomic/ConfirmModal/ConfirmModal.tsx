import React, { FC } from 'react'
import { useIntl } from 'react-intl'
import Button from '../Button'
import Modal from '../Modal'
import { messages as t } from './ConfirmModal.i18n'
import { defaultProps, Props } from './ConfirmModal.types'
import ModalFooter from '../Modal/components/ModalFooter'

const ConfirmModal: FC<Props> = (props) => {
    const { onConfirm, confirmButtonText, cancelButtonText, title, body, loading, show, onClose, data, confirmDisabled, ...rest } = props
    const { formatMessage: _ } = useIntl()

    const renderFooter = (
        <ModalFooter
            right={
                <div className='modal-buttons'>
                    <Button className='modal-button' disabled={loading} onClick={onClose} variant='secondary'>
                        {cancelButtonText || _(t.cancel)}
                    </Button>
                    <Button
                        className='modal-button'
                        disabled={loading || confirmDisabled}
                        loading={loading}
                        onClick={() => onConfirm(onClose, data)}
                        variant='primary'
                    >
                        {confirmButtonText || _(t.confirm)}
                    </Button>
                </div>
            }
        />
    )

    return (
        <Modal
            {...rest}
            closeButton={!loading}
            onClose={!loading ? onClose : () => {}}
            renderBody={body}
            renderFooter={renderFooter}
            show={show}
            title={title}
        />
    )
}

ConfirmModal.displayName = 'ConfirmModal'
ConfirmModal.defaultProps = defaultProps

export default ConfirmModal
