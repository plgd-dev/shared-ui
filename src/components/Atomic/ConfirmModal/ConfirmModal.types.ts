import { ReactNode } from 'react'
import { Props as ModalProps } from '../Modal/Modal.types'

export type Props = {
    body: ReactNode
    cancelButtonText?: string
    children?: ReactNode
    confirmButtonText?: string
    confirmDisabled?: boolean
    data?: {}
    loading?: boolean
    onClose: () => void
    onConfirm: (onClose: () => void, data?: {}) => void
    show: boolean
    title: ReactNode
} & Pick<ModalProps, 'zIndex'>

export const defaultProps = {
    show: false,
    loading: false,
    confirmDisabled: false,
    data: {},
}
