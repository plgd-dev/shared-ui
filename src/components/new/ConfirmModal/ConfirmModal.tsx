import React, { FC } from 'react'
import { useIntl } from 'react-intl'
import Button from '../Button'
import Modal from '../Modal'
import { messages as t } from './ConfirmModal.i18n'
import { defaultProps, Props } from './ConfirmModal.types'

const ConfirmModal: FC<Props> = (props) => {
    const { onConfirm, confirmButtonText, cancelButtonText, title, body, loading, show, onClose, data, confirmDisabled, ...rest } = props
    const { formatMessage: _ } = useIntl()

    const renderFooter = (
        <div className='w-100 d-flex justify-content-end align-items-center'>
            <Button variant='secondary' onClick={onClose} disabled={loading}>
                {cancelButtonText || _(t.cancel)}
            </Button>
            <Button variant='primary' onClick={() => onConfirm(onClose, data)} loading={loading} disabled={loading || confirmDisabled}>
                {confirmButtonText || _(t.confirm)}
            </Button>
        </div>
    )

    return (
        <Modal
            {...rest}
            show={show}
            onClose={!loading ? onClose : () => {}}
            title={title}
            renderBody={body}
            renderFooter={renderFooter}
            closeButton={!loading}
        />
    )
}

ConfirmModal.displayName = 'ConfirmModal'
ConfirmModal.defaultProps = defaultProps

export default ConfirmModal
