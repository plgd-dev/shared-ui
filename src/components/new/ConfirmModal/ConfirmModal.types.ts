import { ReactNode } from 'react'

export type Props = {
    body: ReactNode
    cancelButtonText?: string
    children: ReactNode
    confirmButtonText?: string
    confirmDisabled?: boolean
    data?: {}
    loading?: boolean
    onClose: () => void
    onConfirm: (onClose: () => void, data?: {}) => void
    show: boolean
    title: ReactNode
}

export const defaultProps = {
    show: false,
    loading: false,
    confirmDisabled: false,
    data: {},
}
