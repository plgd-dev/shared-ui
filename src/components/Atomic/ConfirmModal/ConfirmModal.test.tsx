import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import ConfirmModal from './ConfirmModal'

describe('<ConfirmModal>', () => {
    it('renders without crashing', () => {
        const { getByText, asFragment } = render(
            <ConfirmModal
                body={<div>ConfirmModal</div>}
                cancelButtonText='Cancel'
                confirmButtonText='Confirm'
                onClose={() => {}}
                onConfirm={() => {}}
                show={true}
                title='Title'
            />
        )
        expect(getByText('ConfirmModal')).toBeInTheDocument()

        expect(asFragment()).toMatchSnapshot()
    })

    it('calls onClose when cancel button is clicked', () => {
        const onClose = jest.fn()
        const { getByText } = render(
            <ConfirmModal
                body={<div>Body</div>}
                cancelButtonText='Cancel'
                confirmButtonText='Confirm'
                onClose={onClose}
                onConfirm={() => {}}
                show={true}
                title='Title'
            />
        )
        fireEvent.click(getByText('Cancel'))
        expect(onClose).toHaveBeenCalled()
    })

    it('calls onConfirm when confirm button is clicked', () => {
        const onConfirm = jest.fn()
        const { getByText } = render(
            <ConfirmModal
                body={<div>Body</div>}
                cancelButtonText='Cancel'
                confirmButtonText='Confirm'
                onClose={() => {}}
                onConfirm={onConfirm}
                show={true}
                title='Title'
            />
        )
        fireEvent.click(getByText('Confirm'))
        expect(onConfirm).toHaveBeenCalled()
    })

    it('does not call onClose when loading', () => {
        const onClose = jest.fn()
        const { getByText } = render(
            <ConfirmModal
                body={<div>Body</div>}
                cancelButtonText='Cancel'
                confirmButtonText='Confirm'
                loading={true}
                onClose={onClose}
                onConfirm={() => {}}
                show={true}
                title='Title'
            />
        )
        fireEvent.click(getByText('Cancel'))
        expect(onClose).not.toHaveBeenCalled()
    })

    it('does not call onConfirm when loading', () => {
        const onConfirm = jest.fn()
        const { getByText } = render(
            <ConfirmModal
                body={<div>Body</div>}
                cancelButtonText='Cancel'
                confirmButtonText='Confirm'
                loading={true}
                onClose={() => {}}
                onConfirm={onConfirm}
                show={true}
                title='Title'
            />
        )
        fireEvent.click(getByText('Confirm'))
        expect(onConfirm).not.toHaveBeenCalled()
    })

    it('does not call onConfirm when confirmDisabled', () => {
        const onConfirm = jest.fn()
        const { getByText } = render(
            <ConfirmModal
                body={<div>Body</div>}
                cancelButtonText='Cancel'
                confirmButtonText='Confirm'
                confirmDisabled={true}
                onClose={() => {}}
                onConfirm={onConfirm}
                show={true}
                title='Title'
            />
        )
        fireEvent.click(getByText('Confirm'))
        expect(onConfirm).not.toHaveBeenCalled()
    })
})
