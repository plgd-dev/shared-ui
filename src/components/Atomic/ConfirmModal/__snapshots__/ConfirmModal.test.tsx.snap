// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmModal> renders without crashing 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: rgba(25, 26, 26, 0.9);
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.emotion-2 {
  background: #FFFFFF;
  padding: 24px 0;
  min-width: 600px;
  max-width: 90%;
  max-height: 90vh;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-bottom: 2px solid #E6E9ED;
  padding: 12px 24px 32px 24px;
}

.emotion-4 {
  color: #0A2965;
  font-family: 'Circular Pro',sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 24px;
  line-height: 30px;
  letter-spacing: -0.5px;
  margin: 0;
  font-family: 'Circular Pro',sans-serif;
  font-style: normal;
  font-weight: 700;
  font-size: 24px;
  line-height: 30px;
  letter-spacing: -0.5px;
  color: #0A2965;
  max-width: 100%;
}

@media (max-width: 767px) {
  .emotion-4 {
    font-size: 20px;
    line-height: 24px;
  }
}

.emotion-5 {
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 160%;
  color: #81868C;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
}

.emotion-5 span {
  margin-right: 10px;
}

.emotion-5:hover {
  color: #2261AE;
}

.emotion-6 {
  background: #FFFFFF;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-y: auto;
  overflow-x: hidden;
  padding-right: 24px;
  padding-left: 24px;
  padding-top: 24px;
}

.emotion-7 {
  padding: 24px 24px 0 24px;
  background: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-7 .modal-buttons {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 -8px;
}

.emotion-7 .modal-buttons .modal-button {
  margin: 0 4px;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
}

.emotion-9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 600;
  text-align: center;
  border-radius: 8px;
  white-space: nowrap;
  box-sizing: border-box;
  cursor: pointer;
  background: #ffffff;
  border: 1px solid #0A2965;
  color: #0A2965;
  padding: 10px 24px;
  font-size: 14px;
  line-height: 16px;
  height: 48px;
}

.emotion-9:hover {
  background: #0A2965;
  border-color: #0A2965;
  color: #FFFFFF;
}

.emotion-10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 600;
  text-align: center;
  border-radius: 8px;
  white-space: nowrap;
  box-sizing: border-box;
  cursor: pointer;
  background: #2261AE;
  border: 1px solid #2261AE;
  color: #FFFFFF;
  padding: 10px 24px;
  font-size: 14px;
  line-height: 16px;
  height: 48px;
}

.emotion-10:hover {
  background: #1D589F;
  border-color: #1D589F;
  color: #FFFFFF;
}

<div
    class="emotion-0"
    style="opacity: 1;"
  >
    <div
      class=" emotion-1"
      role="dialog"
      style="opacity: 1; transform: none;"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <div>
            <h4
              class="h4 emotion-4"
            >
              Title
            </h4>
          </div>
          <a
            class="emotion-5"
            href="#"
          >
            <span>
              Close
            </span>
             
            <svg
              fill="none"
              height="26"
              viewBox="0 0 32 32"
              width="26"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 29.333c7.333 0 13.333-6 13.333-13.333S23.333 2.667 16 2.667 2.667 8.667 2.667 16s6 13.333 13.333 13.333M12.227 19.773l7.546-7.546M19.773 19.773l-7.546-7.546"
                stroke="currentcolor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
          </a>
        </div>
        <div
          class="emotion-6"
        >
          <div>
            ConfirmModal
          </div>
        </div>
        <div
          class="emotion-7"
        >
          <div
            class="emotion-8"
          >
            <div />
            <div
              class="modal-buttons"
            >
              <button
                class="modal-button emotion-9"
                tabindex="1"
                type="submit"
              >
                Cancel
              </button>
              <button
                class="modal-button emotion-10"
                tabindex="1"
                type="submit"
              >
                Confirm
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
